### ISSUE TITLE
Corrections to figures

### ISSUE THREAD

**Issue by teorth** [km-git-acc/dbn_upper_bound#107]
I guess I should use this issues forum rather than email for requests for corrections to the figures.  Here's what I have so far:

* For the left figure in Figure 2, the horizontal axis should be log_{10} x rather than 10^x.

* For figure 5, it should be |f_t + RS| rather than |f_t + RS|.  Actually I've just tweaked the notation in the paper so that one can use the more precise description |f_t - C_t/B_t|.

* To complement Figures 3,4, it would also be nice to have a figure displaying what happens at significantly larger values of x (e.g. x ~ 10^5 or x ~ 10^6, if numerically feasible) to illustrate how the accuracy of the f_t approximation to H_t/B_t improves with x.

**Comment by km-git-acc** [comment]
@teorth @rudolph-git-acc 
I have fixed the labeling in Figs 2 and 5 and uploaded the edited figures and files to the writeup folder. 
We will also check on the 3rd point. I think the bottleneck was that H_t is slow to compute at large x (for eg. 5 sec for x near 10^5 and 90 sec for x near 10^6) so we went for the easier option, but atleast the x near 10^5 graph could be done in a few hours.

**Comment by rudolph-git-acc** [comment]
@teorth @km-git-acc 

Oops. It seems we did some work in parallel here and was just planning to post the corrected graphs here as well. Have managed to get plots for x~10^6. Just waiting for the final one to complete. Will post shortly.

**Comment by teorth** [comment]
These look great, thanks!

I may have set up an edit conflict as I was also editing the intro.tex file without first merging from master; I am not sure how to fix it from my end but perhaps the merge can be completed from the master end.

Looking at the wide fluctuation of H_t/B_t I wonder if it is also worth having a separate plot comparing H_t/B_t with the partial Euler products that we use to select the location of the barrier, perhaps re-using some of the ranges of x,y,t in one of the above plots for sake of comparison.  Certainly these figures make it clear that the choice of barrier location can be important!

**Comment by rudolph-git-acc** [comment]
Yes, visualizing the impact of the partial Euler product is a great idea! We'll work this further.

To avoid editing conflicts, it is probably easiest when you simply ignore the recent changes we made and just finish your detailed walkthrough of the write-up. We can then easily re-include all the images and tables at the end of the process (and also ensure the formatting and positioning works correctly).

**Comment by km-git-acc** [comment]
Have resolved the merge conflict, and also updated the pdf with the above figures. Rudolph's suggestion is a good one, in that we can now wait for all the tex edits to finish first.

**Comment by km-git-acc** [comment]
@teorth @rudolph-git-acc 

Below are two graphs comparing the f<sub>t</sub> approximation with partial euler products (t=0,y=1) upto the first 30 primes (the first one near X=6x10^10+83951.5, and the second one near X=10^6).
As the number of primes in the euler product increases, the fit with f<sub>t</sub> improves significantly as expected.

<img width="424" alt="approximation with partial euler products near 6x10e10 83951 5" src="https://user-images.githubusercontent.com/35831522/45932239-0df61c80-bf97-11e8-9fcf-43de311fab17.png">

<img width="424" alt="approximation with partial euler products near 10e6" src="https://user-images.githubusercontent.com/35831522/45932243-177f8480-bf97-11e8-9eea-c02247d5fa9d.png">

Is this in the right direction, and are there any pointers to bring the plot style/formatting more in line with the paper?

**Comment by km-git-acc** [comment]
I made the changes and a few others, and have added the below chart to the writeup.
<img width="423" alt="euler_product_approximation" src="https://user-images.githubusercontent.com/35831522/45962543-b8754a80-c03e-11e8-8962-4f21914c2dee.png">
Exact figure placement with Latex seems a bit tricky, and currently the chart is appearing on pg. 37 after Section 8 starts.
p1,p2,.. labels have been replaced by ep<sub>n</sub>(x) = eulerprod(x,p<sub>n</sub>)

Also, the number 27 used in the eulerprod definition is actually 29, since the first 10 primes were used for calculations. Have changed that as well.

**Comment by teorth** [comment]
Here's another possibility for a figure in the introduction: a comparision between the error |f_t - H_t/B_t| and the error bound e_A + e_B + e_{C,0} (as well as the component factors e_A, e_B, e_{C,0}).  Probably the most informative graphic will have x vary over a long range, e.g. using log-scale for x, then the error should oscillate wildly (similar to the left figure in Figure 2) but the error bounds e_A, e_B, e_{C,0} should be much smoother.  (But one could also have a closeup picture near a place where N gets incremented to highlight the discontinuity present there.)

**Comment by rudolph-git-acc** [comment]
@teorth @km-git-acc 

Yes, such visuals about the error terms are definitely doable. I will start working these!

**Comment by teorth** [comment]
Thanks!  Another image request: the asymptotic analysis in Theorem 1.5 suggests that the zeroes of H_t eventually approach the points where \frac{X}{4\pi} \log \frac{X}{4\pi} - \frac{X}{4\pi} + \frac{11}{8} + \frac{t}{16} \log \frac{X}{4\pi} is an integer.  I don't know if you have the capability to locate zeroes of H_t for large enough x that one can see this "solidifying" of the zeroes, but it would be nice to have a graphic at large x showing this (and perhaps also a graphic at small x showing the more disordered behaviour of zeroes).

**Comment by rudolph-git-acc** [comment]
@teorth @km-git-acc 

Making good progress on visualising the error bounds. Jobs are running now and expect to post the graphs later today.

We have developed a real-root finder that works well up till at least the x=10^12 domain (it is based on the ABC_eff proxy for Ht). Just ran a few tests and the integer-effect is very clearly there. It gets stronger with increasing x and also with increasing t. Will aim to produce a visual for that tomorrow.

**Comment by rudolph-git-acc** [comment]
@teorth @km-git-acc 

Quick update: during the last two weeks, KM and I have been progressing the numerical work through using grid-computing (Boinc) to conditionally proof a DBN <= 0.11 and <= 0.10. After being listed on the Boinc-homepage, we saw a massive increase in the number of volunteering participants, with some bringing extremely powerful computers to this great collaborative effort. 

Both DBN-cases have a Triangle bound that is already positive at the Barrier location, so no further numerical work is required beyond the verification that no zeros have passed the Barrier itself. We probably need less than a week to fully complete both.  

Please let us know whether any additional visuals/tables/texts need to be produced for the write-up on top of the ones in the previous posts.

**Comment by teorth** [comment]
Great!  I managed to miss being notified of your previous update, but the error bound pictures look great, feel free to insert them into the writeup (I can edit them later if needed).

Probably the reason that f_t - H_t / B_t is not too wild is that it is being dominated by the "C" term (compare for instance the oscillations with the plot of C_0 in Figure 6 of the writeup.

One other graphic that came to mind relates to Figure 2 of the writeup.  In addition to a plot of the relative magnitude |H_t|/|B_t| vs log x on the left of Figure 2, one could also add a plot of log |H_t| and log |B_t| vs log x to illustrate how B_t captures the bulk of the exponential decay of H_t.  (Also it illustrates how more naive ways to calculate H_t would need extremely high numerical precision at large x due to how small H_t is.)

**Comment by teorth** [comment]
Looks great!  Please do include the previous figures also.

**Comment by rudolph-git-acc** [comment]
@teorth @km-git-acc 

All done :) Not all graphs are well positioned yet, but probably best to wait until the write-up is complete and then 'force' them into the right location.

**Comment by km-git-acc** [comment]
@teorth @rudolph-git-acc 
As an update, recently using the grid computing setup, we computed winding numbers in the barrier region,
X=9*10^21 + [70686,70687], y=[0.11832,1], t = [0,0.093],
and all the winding numbers turned to be zero (around 60000 t-steps),
(https://github.com/km-git-acc/dbn_upper_bound/blob/master/output/windingnumbers/windnum_nolemma_x9000000000000000070686_y_0.11832_1_t_0_0.093.txt)
giving dbn < 0.1 conditionally.

**Comment by teorth** [comment]
Hi, sorry for not being around the past few weeks, it's been crazily busy around here.

I'm going through the writeup again, hopefully to get it to a near-final form (even if the numerics keep going, I think activity on the theory side has basically ceased and so it would be good to wrap things up).  I'm looking again at Figure 3 (the last figure Rudolph provided here) and wonder if it may look better if we plot log|H_t| and log |B_t| against x rather than log x, for the same range of x (1 to 10^7).  Then the plot should be roughly linear.  Either way it does convey though that H_t gets incredibly tiny for large x!

**Comment by rudolph-git-acc** [comment]
Here is an update version indeed showing an almost linear relation. Happy to make any other changes.

![log_ht and log_bt vs x](https://user-images.githubusercontent.com/36714969/48707835-fc309e80-ec00-11e8-9475-9a533f6869a6.png)

**Comment by teorth** [comment]
Thanks!  I have incorporated this figure into the latest writeup.  On a related note, it occurs to me that for Figure 2, a plot of |H_t/B_t - 1| is slightly more informative than a plot of |H_t/B_t| since the latter does not preclude the scenario in which H_t differs from B_t by a large phase (e.g. for instance if H_t was close to -B_t).   One could in principle also do this for Figure 4, 5 except that for the right-hand figure the residual H_t/B_t - f_t would presumably be visually indistinguishable from zero (and one would have to rescale the axes), and we do something like this anyway for Figure 6, so perhaps we could leave those as is.

**Comment by rudolph-git-acc** [comment]
Below is the new version of Figure 2 now with |H_t/B_t - 1| :

![image](https://user-images.githubusercontent.com/36714969/48727355-49793400-ec32-11e8-9509-8cde1c113a04.png)

**Comment by teorth** [comment]
Great!  I note that the description of the right-hand image (both above the image and in the box) still has |H_t|/|B_t| rather than |H_t/B_t-1|, is this a typo?

**Comment by rudolph-git-acc** [comment]
Ah, not a typo. I had actually fixed it for the right hand graph only. Will do both!

**Comment by rudolph-git-acc** [comment]
Here is the updated version, now also done the right-hand graph:

![image](https://user-images.githubusercontent.com/36714969/48729244-75e37f00-ec37-11e8-9d4c-a550c547fb96.png)

**Comment by teorth** [comment]
Great! I now put this into the writeup.

**Comment by teorth** [comment]
Finally got some time to work on the writeup again.  I think there may be a typo in what is currently Figure 9 (Integerconvergence_t_and_x_direction.png) on the writeup, in that the x-axis for the left figure should be t rather than log_{10} x.  The original version of the figure (the third picture in Rudolph's Sep 25 comment) has the axis labeled correctly though.  Would it be possible to fix the axis labeling?  Thanks.

**Comment by rudolph-git-acc** [comment]
Good catch! This was the result of a stupid copy/paste error. Here is the updated version. Please let us know how we can further help with the write-up.

![integerconvergence_t_and_x_direction](https://user-images.githubusercontent.com/36714969/50187335-16ec5500-031e-11e9-9722-e17a4b45bc9a.png)
